src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php

```diff
@@ -40,25 +43,28 @@
  */
 final class EagerLoadingExtension implements ContextAwareQueryCollectionExtensionInterface, QueryItemExtensionInterface
 {
+    use ContextTrait;
     use EagerLoadingTrait;
 
     private $propertyNameCollectionFactory;
     private $propertyMetadataFactory;
     private $classMetadataFactory;
     private $maxJoins;
-    private $serializerContextBuilder;
+    private $serializerContextFactory;
     private $requestStack;
 
     /**
+     * @param SerializerContextBuilderInterface|SerializerContextFactoryInterface $serializerContextBuilder
+     *
      * @TODO move $fetchPartial after $forceEager (@soyuka) in 3.0
      */
-    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, ResourceMetadataFactoryInterface $resourceMetadataFactory, int $maxJoins = 30, bool $forceEager = true, RequestStack $requestStack = null, SerializerContextBuilderInterface $serializerContextBuilder = null, bool $fetchPartial = false, ClassMetadataFactoryInterface $classMetadataFactory = null)
+    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, ResourceMetadataFactoryInterface $resourceMetadataFactory, int $maxJoins = 30, bool $forceEager = true, RequestStack $requestStack = null, /* SerializerContextBuilderInterface $serializerContextFactory */$serializerContextBuilder = null, bool $fetchPartial = false, ClassMetadataFactoryInterface $classMetadataFactory = null)
     {
-        if (null !== $this->requestStack) {
+        if (null !== $requestStack) {
             @trigger_error(sprintf('Passing an instance of "%s" is deprecated since version 2.2 and will be removed in 3.0. Use the data provider\'s context instead.', RequestStack::class), E_USER_DEPRECATED);
         }
-        if (null !== $this->serializerContextBuilder) {
-            @trigger_error(sprintf('Passing an instance of "%s" is deprecated since version 2.2 and will be removed in 3.0. Use the data provider\'s context instead.', SerializerContextBuilderInterface::class), E_USER_DEPRECATED);
+        if (null !== $serializerContextBuilder) {
+            @trigger_error(sprintf('Passing an instance of "%s" or "%s" is deprecated since version 2.2 and will be removed in 3.0. Use the data provider\'s context instead.', SerializerContextBuilderInterface::class, SerializerContextFactoryInterface::class), E_USER_DEPRECATED);
```

Comment:
We don't need to add support for the new `SerializerContextFactoryInterface` as passing a context factory is deprecated anyway and shouldn't be done. I think you can remove the changes in this file.

Reply:
They were some bugs with the deprecations though (that's why the `$this` has been removed).